components/movementsensor/wheeledodometry/wheeled_odometry.go

```diff
@@ -0,0 +1,311 @@
+	return movementsensor.Readings(ctx, o, extra)
+}
+
+func (o *odometry) Accuracy(ctx context.Context, extra map[string]interface{}) (map[string]float32, error) {
+	return map[string]float32{}, movementsensor.ErrMethodUnimplementedAccuracy
+}
+
+func (o *odometry) Properties(ctx context.Context, extra map[string]interface{}) (*movementsensor.Properties, error) {
+	return &movementsensor.Properties{
+		LinearVelocitySupported:     true,
+		AngularVelocitySupported:    true,
+		OrientationSupported:        true,
+		PositionSupported:           true,
+		LinearAccelerationSupported: false,
+	}, nil
+}
+
+func (o *odometry) Close(ctx context.Context) error {
+	o.cancelFunc()
+	o.activeBackgroundWorkers.Wait()
+	return nil
+}
+
+func (o *odometry) trackPosition(ctx context.Context) {
+	o.activeBackgroundWorkers.Add(1)
+	utils.PanicCapturingGo(func() {
+		defer o.activeBackgroundWorkers.Done()
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+
+			// always use the first pair until more than one pair of motors is supported in this model
+			leftMove, err := o.motors[0].left.IsMoving(ctx)
+			if err != nil {
+				o.logger.Error(err)
+			}
+			rightMove, err := o.motors[0].right.IsMoving(ctx)
+			if err != nil {
+				o.logger.Error(err)
+			}
+			if !leftMove && !rightMove {
+				// neither motors are moving, so the odometry values do not need to be updated
+				continue
+			}
+			// sleep for the desired time interval
+			time.Sleep(time.Duration(o.timeIntervalMSecs) * time.Millisecond)
+
+			// use GetInParallel to ensure the left and right motors are polled at the same time
+			fs := []rdkutils.FloatFunc{}
+			// always use the first pair until more than one pair of motors is supported in this model
+			fs = append(fs, func(ctx context.Context) (float64, error) { return o.motors[0].left.Position(ctx, nil) })
+			fs = append(fs, func(ctx context.Context) (float64, error) { return o.motors[0].right.Position(ctx, nil) })
+
+			_, positions, err := rdkutils.GetInParallel(ctx, fs)
+			if err != nil {
+				o.logger.Error(err)
+				continue
+			}
+
+			// current position of the left and right motors in revolutions
+			left := positions[0]
+			right := positions[1]
+
+			// difference in the left and right motors since the last iteration, in mm
+			leftDist := (left - o.lastLeftPos) * o.wheelCircumference
+			rightDist := (right - o.lastRightPos) * o.wheelCircumference
+
+			// update lastLeftPos and lastRightPos to be the current position in mm
+			o.lastLeftPos = left
+			o.lastRightPos = right
+
+			// linear and angular distance the center point has traveled
+			centerDist := (leftDist + rightDist) / 2
+			centerAngle := (rightDist - leftDist) / o.baseWidth
+
+			// update the position and orientation values accordingly
+			o.orientation.Yaw += centerAngle
+
+			// limit the orientation to the range 0 - 360 degrees
+			o.orientation.Yaw = math.Mod(o.orientation.Yaw, 2*math.Pi)
```

Comment:
@randhid reverting this because I can't figure out a way to use mod and not just get negative angles (such as -90 when I want 270)

Reply:
I can do -180 to 180 if that's what we want, I don't have a strong opinion

Reply:
the code I linked goes 0-360, I don't have a strong opinion, and orientation will cast it to -180 to 180 in our api call anyway.